packages/xo-server/src/xo-mixins/resource-sets.mjs

```diff
@@ -444,6 +444,9 @@ export default class {
 
   async shareVmResourceSet(vmId) {
     const xapi = this._app.getXapi(vmId)
+    await xapi.barrier(xapi.getObject(vmId).$ref)
+    // barrier function is used to ensure completion of previous asynchronous operations
+    // on the vmId object, guaranteeing that all modifications are considered before retrieving the resourceSetId value.
```

Comment:
If I understand correctly, the issue we are fixing here is that `shareVmResourceSet` is called right after `setVmResourceSet` in `vm.set`.

The problem should therefore be fixed there and not here.

Otherwise, we might as well never use `getObject()` or always use `barrier()` which is not what we want ^^